docs/source/authoring_flows.rst

```diff
@@ -549,19 +722,291 @@ View the `Move (copy and delete) files using Globus`_ in the Globus web app.
+                    "is_folder.=": "getattr('DestinationPathInfo.details.DATA[0].is_folder', False)",
+                    "destination_file.=": "getattr('DestinationPathInfo.details.DATA[0].name', '/')",
+                    "destination_folder.=": "DestinationPathInfo.details.path"
+                },
+                "ResultPath": "$.DestinationInfo"
+            },
+            "CreateActionInputs": {
+                "Next": "MakeIntermediateDir",
+                "Type": "ExpressionEval",
+                "Comment": "Setup the inputs for all the Actions to perform the two stage transfer",
+                "Parameters": {
+                    "MakeDirInput": {
+                        "path.=": "intermediate.path + '/' + _context.run_id",
+                        "endpoint_id.$": "$.intermediate.id"
+                    },
+                    "Transfer1Input": {
+                        "label.=": "getattr('transfer1_label', 'Stage One Transfer for Flow Run with id ' + _context.run_id)",
+                        "transfer_items": [
+                            {
+                                "recursive.$": "$.SourceInfo.is_recursive",
+                                "source_path.$": "$.source.path",
+                                "destination_path.=": "intermediate.path + '/' + _context.run_id + '/' + SourceInfo.source_file"
+                            }
+                        ],
+                        "source_endpoint_id.$": "$.source.id",
+                        "destination_endpoint_id.$": "$.intermediate.id"
+                    },
+                    "Transfer2Input": {
+                        "label.=": "getattr('transfer2_label', 'Stage Two Transfer for Flow Run with id ' + _context.run_id)",
+                        "transfer_items": [
+                            {
+                                "recursive.=": "SourceInfo.is_recursive",
+                                "source_path.=": "intermediate.path + '/' + _context.run_id + '/' + SourceInfo.source_file",
+                                "destination_path.=": "(destination.path + '/' + SourceInfo.source_file) if (DestinationInfo.is_folder or ((not DestinationInfo.exists) and SourceInfo.is_recursive)) else destination.path"
+                            }
+                        ],
+                        "source_endpoint_id.$": "$.intermediate.id",
+                        "destination_endpoint_id.$": "$.destination.id"
+                    },
+                    "DeleteFromIntermediateInput": {
+                        "items.=": "[(intermediate.path + '/' + _context.run_id)]",
+                        "label.=": "getattr('delete_intermediate_label', 'Delete from Intermediate for Flow Run with id ' + _context.run_id)",
+                        "recursive": true,
+                        "endpoint_id.$": "$.intermediate.id"
+                    }
+                },
+                "ResultPath": "$.ActionInputs"
+            },
```

Comment:
I'm not convinced this state helps with legibility more than it hurts.

Seems like the idea was to take all of the complexity for the next 4 states & bundle them into one.
Could we just move each `$.ActionInputs.*Input` to the state which uses that input? As best as I can tell we're not re-using any of the data generated here, so it's just de-localizing it.